### ISSUE TITLE
macOS screen size not finding correct size

### ISSUE THREAD

**Issue by maddox05** [asweigart/pyautogui#808]
When calling pyautogui.size for Mac's with the Retina display, it says the wrong size,around half of what the size is. This causes locate on screen and mouse move to break.

**Comment by ghost** [comment]
Could you provide your Mac's actual screens size along with what PyAutoGUI claims it to be? There's a few reasons this could be that actually aren't unique to PyAutoGUI, many graphics/UI related tools struggle with this sort of issue on retina displays.
One possibility is that this is caused by retina displays having higher pixel density. This means that for each point on the screen ("logical pixel") there might be 2x2 physical pixels present. This corresponds to what you're describing as the program recognizing the screen as being half the size that it actually is.

As of September 25, 2023, I'll be trying to address this issue by creating a work around for retina display users. If someone could assign this ticket to me that would be appreciated.

**Comment by maddox05** [comment]
yeah its the macbook pro with a retina display. Good explanation, for whatever reason dividing the values by 2 seems to fix going to a point incorrectly, it read the screen half of what Apple "says" it is. The actual size from Apple is 3024 × 1964 and pyautogui.screen() halfs both values.
